src/index.ts

```diff
@@ -583,4 +585,35 @@ export class ReadonlyArrayType<RT extends Any> implements Type<ReadonlyArray<Typ
 export const readonlyArray = <RT extends Any>(type: RT, name?: string): ReadonlyArrayType<RT> =>
   new ReadonlyArrayType(type, name)
 
+export class StrictType<RT extends InterfaceType<any>> implements Type<TypeOf<RT>> {
+  readonly _tag: 'StrictType' = 'StrictType'
+  readonly _A: TypeOf<RT>
+  readonly validate: Validate<TypeOf<RT>>
+  constructor(readonly type: RT, readonly name: string = `StrictType<${type.name}>`) {
+    const len = Object.keys(type.props).length
+    this.validate = (v, c) =>
+      type.validate(v, c).chain(o => {
+        const keys = Object.keys(v)
+        if (keys.length !== len) {
+          const errors: Errors = []
+          for (let i = 0; i < keys.length; i++) {
+            const key = keys[i]
+            if (!type.props.hasOwnProperty(key)) {
+              errors.push(getValidationError(o[key], c.concat(getContextEntry(key, never))))
+            }
+          }
+          if (errors.length) {
+            return failures(errors)
+          }
+        }
+        return success(o)
+      })
+  }
+}
+
+/** Specifies that only the given interface properties are allowed */
+export function strict<RT extends InterfaceType<any>>(type: RT, name?: string): StrictType<RT> {
+  return new StrictType(type, name)
+}
```

Comment:
Because strict only works with interfaces anyways, i think it might be clearer (and more compact) for this function to directly take Props as an argument instead of an interface.

```ts
export function strict<P extends Props>(props: P, name?: string): StrictType<InterfaceType<P>> {
  return new StrictType(type(props), name)
}
```

Reply:
Agreed, also is more consistent with the `partial` combinator